### PR TITLE
Introduce rename operation

### LINKED ISSUE #50: Renaming virtual machines
Page: Virtual Machines

It would be nice to have the option to rename existing virtual machines in order to better distinguish between multiple VMs on the same host.

### PR DESCRIPTION
**Pull request by KKoukiou**
Fixes #50

## Machines: Add support for renaming VMs

The Machines page can rename virtual machines.

Screen Shot 2021-09-08 at 18 45 02
Screen Shot 2021-09-08 at 18 43 17

### CODE REVIEW

**Comment by garrett** [comment]
Cannot rename VMs that have a managed save state:

Screenshot 2021-09-09 at 14-30-48 Virtual Machines - garrett Bolt

Perhaps we should handle this like such:
- Context menu "Rename" disabled
- Hovering the disabled item would show "Cannot rename virtual machines with a managed save state"

WDYT?

---

The dialog lets you rename to the same name, but throws an error. Should we handle this by disabling the rename action and/or showing some error text below?

We should probably prefill the entry with the existing name, so someone can easily alter the name of the VM. But greeting them with an error by default isn't a great idea, so at least an initial name check (when the dialog appears) shouldn't trigger an error.

Screenshot 2021-09-09 at 14-33-21 Virtual Machines - garrett Bolt


---

Invalid characters throws an "XML error".

Screenshot 2021-09-09 at 14-36-31 Virtual Machines - garrett Bolt

Weirdly enough, you can name things not just with spaces but `<`, `>`, `!`, `,`, ` ` and other characters. :shrug: It seems to work fine. So I guess the issue is only with slashes?

File: src/components/vm/vmRenameDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply src/components/vm/vmRenameDialog.jsx:4]: Yes. :)

**Comment by KKoukiou** [comment]
As agreed on IRC we will not special case the errors for inline pre-validation of the dialog - rather let libvirt show them - as they do in the screenshots above.

**Review by martinpitt — APPROVED** [review]
This looks nice, thanks! Seems the points above were already settled.

**Pull request merged**